### PR TITLE
add odt rpga feather

### PR DESCRIPTION
**Pull request by skerr92**
Adds a CircuitPython build for RPGA Feather. PID is under review with pid.codes. I've tested this on my boards and it works as expected.

### CODE REVIEW

File: ports/raspberrypi/boards/odt_rpga_feather/pins.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    { MP_ROM_QSTR(MP_QSTR_F_CSN), MP_ROM_PTR(&pin_GPIO13) },
+
+    { MP_ROM_QSTR(MP_QSTR_D21), MP_ROM_PTR(&pin_GPIO21) },
+    { MP_ROM_QSTR(MP_QSTR_D22), MP_ROM_PTR(&pin_GPIO22) },
+    { MP_ROM_QSTR(MP_QSTR_DD23), MP_ROM_PTR(&pin_GPIO23) },
+    { MP_ROM_QSTR(MP_QSTR_D24), MP_ROM_PTR(&pin_GPIO24) },
+    { MP_ROM_QSTR(MP_QSTR_D25), MP_ROM_PTR(&pin_GPIO25) },
+    { MP_ROM_QSTR(MP_QSTR_LED), MP_ROM_PTR(&pin_GPIO25) },
+
+    { MP_ROM_QSTR(MP_QSTR_I2C), MP_ROM_PTR(&board_i2c_obj) },
```
Comment (dhalbert) [inline ports/raspberrypi/boards/odt_rpga_feather/pins.c:65]: If you have `STEMMA_SCL` and `STEMMA_SDA`, how about `STEMMA_I2C`? There are a number of boards with that. See one for how to specify multiple sets of I2C board objects.

**Review by dhalbert — CHANGES_REQUESTED** [review]
One comment re possible `board.STEMMA_I2C()`

File: ports/raspberrypi/boards/odt_rpga_feather/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#define MICROPY_HW_MCU_NAME "rp2040"
+
+#define MICROPY_HW_LED_STATUS (&pin_GPIO25)
+
+#define DEFAULT_I2C_BUS_SCL (&pin_GPIO10)
+#define DEFAULT_I2C_BUS_SDA (&pin_GPIO11)
+
+#define CIRCUITPY_BOARD_I2C         (2)
+#define CIRCUITPY_BOARD_I2C_PIN     {{.scl = &pin_GPIO10, .sda = &pin_GPIO11}, \
+                                     {.scl = &pin_GPIO8, .sda = &pin_GPIO9}}
```
Comment (dhalbert) [inline ports/raspberrypi/boards/odt_rpga_feather/mpconfigboard.h:19]: This has SCL/SDA as 10/11 and 8/9, but pins.c has the opposite.

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/raspberrypi/boards/odt_rpga_feather/pins.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    { MP_ROM_QSTR(MP_QSTR_A1), MP_ROM_PTR(&pin_GPIO27) },
+    { MP_ROM_QSTR(MP_QSTR_D28), MP_ROM_PTR(&pin_GPIO28) },
+    { MP_ROM_QSTR(MP_QSTR_A2), MP_ROM_PTR(&pin_GPIO28) },
+    { MP_ROM_QSTR(MP_QSTR_A29), MP_ROM_PTR(&pin_GPIO29) },
+    { MP_ROM_QSTR(MP_QSTR_D3), MP_ROM_PTR(&pin_GPIO29) },
+
+    { MP_ROM_QSTR(MP_QSTR_SCK), MP_ROM_PTR(&pin_GPIO2) },
+    { MP_ROM_QSTR(MP_QSTR_MOSI), MP_ROM_PTR(&pin_GPIO3) },
+    { MP_ROM_QSTR(MP_QSTR_MISO), MP_ROM_PTR(&pin_GPIO0) },
+    { MP_ROM_QSTR(MP_QSTR_CSN), MP_ROM_PTR(&pin_GPIO1) },
```
Comment (dhalbert) [inline ports/raspberrypi/boards/odt_rpga_feather/pins.c:26]: What is `CSN`? Did you mean `CS` ??

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/raspberrypi/boards/odt_rpga_feather/pins.c (hunk 3 above)
Comment (skerr92) [reply ports/raspberrypi/boards/odt_rpga_feather/pins.c]: no, I meant CSN but for the sake of consistency with other boards it'll be CS. the F_CSN is intentional though as it's how the pin is labeled on the FPGA.

File: ports/raspberrypi/boards/odt_rpga_feather/mpconfigboard.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#define MICROPY_HW_BOARD_NAME "Oak Dev Tech RPGA Feather"
+#define MICROPY_HW_MCU_NAME "rp2040"
+
+#define MICROPY_HW_LED_STATUS (&pin_GPIO25)
+
+#define DEFAULT_I2C_BUS_SCL (&pin_GPIO10)
+#define DEFAULT_I2C_BUS_SDA (&pin_GPIO11)
+
+#define CIRCUITPY_BOARD_I2C         (2)
+#define CIRCUITPY_BOARD_I2C_PIN     {{.scl = &pin_GPIO10, .sda = &pin_GPIO11}, \
```
Comment (dhalbert) [inline ports/raspberrypi/boards/odt_rpga_feather/mpconfigboard.h:18]: 10/11 SCL/SDA is still reversed from what's in pins.c.

**Review by dhalbert — CHANGES_REQUESTED** [review]

**Comment by dhalbert** [comment]
Of course make a PR to https://github.com/adafruit/circuitpython-org now.

**Pull request merged**